src/rosdistro/manifest_provider/git.py

```diff
@@ -53,35 +53,50 @@ def git_manifest_provider(_dist_name, repo, pkg_name):
         raise RuntimeError('Unable to fetch package.xml: %s' % e)
 
 
+def _git_version_gte(version):
+    global _git_client_version
+    if not _git_client_version:
+        cmd = [_git_client_executable, '--version']
+        result = _run_command(cmd)
+        _git_client_version = result['output'].split()[-1]
+    return _git_client_version.split('.') >= version.split('.')
+
+
 def _get_package_xml(url, tag):
     base = tempfile.mkdtemp('rosdistro')
     try:
-        # git 1.7.9 does not support cloning a tag directly, so doing it in two steps
         assert _git_client_executable is not None, "'git' not found"
-        cmd = [_git_client_executable, 'clone', url, base]
-        result = _run_command(cmd, base)
-        if result['returncode'] != 0:
-            raise RuntimeError('Could not clone repository "%s"' % url)
+        if _git_version_gte('1.9.1'):
```

Comment:
I ran `man git clone` on other machine and in the paragraph of `--branch` it mentions something like this:

    --branch can also take tags and detaches the HEAD at that commit in the resulting repository.

The earliest minor version where this was available is 1.8 (see https://git.kernel.org/cgit/git/git.git/tree/Documentation/git-clone.txt?h=v1.8.0#n158).